watchdog/watchdog.go

```diff
@@ -10,34 +10,29 @@ import (
 	"github.com/TwinProduction/gatus/config"
 	"github.com/TwinProduction/gatus/core"
 	"github.com/TwinProduction/gatus/metric"
+	"github.com/TwinProduction/gatus/storage"
 )
 
 var (
-	serviceStatuses = make(map[string]*core.ServiceStatus)
-
-	// serviceStatusesMutex is used to prevent concurrent map access
-	serviceStatusesMutex sync.RWMutex
+	store = storage.NewInMemoryStore()
 
 	// monitoringMutex is used to prevent multiple services from being evaluated at the same time.
 	// Without this, conditions using response time may become inaccurate.
 	monitoringMutex sync.Mutex
 )
 
 // GetJSONEncodedServiceStatuses returns a list of core.ServiceStatus for each services encoded using json.Marshal.
-// The reason why the encoding is done here is because we use a mutex to prevent concurrent map access.
 func GetJSONEncodedServiceStatuses() ([]byte, error) {
-	serviceStatusesMutex.RLock()
+	serviceStatuses := store.GetAll()
 	data, err := json.Marshal(serviceStatuses)
-	serviceStatusesMutex.RUnlock()
 	return data, err
 }
 
 // GetUptimeByServiceGroupAndName returns the uptime of a service based on its group and name
 func GetUptimeByServiceGroupAndName(group, name string) *core.Uptime {
 	key := fmt.Sprintf("%s_%s", group, name)
-	serviceStatusesMutex.RLock()
+	serviceStatuses := store.GetAll()
```

Comment:
I could, but it would be entirely unused at this point.

Is this something we could add _when it's required_ rather than before it's consumed? Otherwise we're kind of imagining a use for it and it's possible we get it wrong anyway.

Reply:
What do you mean by it'd be entirely unused?
It's used to create a badge.

Reply:
Ah, I must've missed that. Badges are very new so when I rebased it seems I skipped over it. Fair point. Ill look into it

Reply:
Yeah I added it while you were working on #32's implementation I believe :P https://github.com/TwinProduction/gatus#uptime-badges

Reply:
add func to store for getting single service status

Reply:
4d24a4d